boa/src/exec/mod.rs

```diff
@@ -256,17 +263,17 @@ impl Interpreter {
     pub(crate) fn to_primitive(
         &mut self,
         input: &mut Value,
-        preferred_type: Option<&str>,
+        preferred_type: Option<PreferredType>,
     ) -> Value {
         let mut hint: &str;
         match (*input).deref() {
             ValueData::Object(_) => {
                 hint = match preferred_type {
                     None => "default",
                     Some(pt) => match pt {
-                        "string" => "string",
-                        "number" => "number",
-                        _ => "default",
+                        PreferredType::String => "string",
+                        PreferredType::Number => "number",
+                        PreferredType::Default => "default",
```

Comment:
Maybe we should make `ordinary_to_primitive` method take in `PreferredType` so we don't have to convert the enum to `&str`.